helpers/htmlAttributeObserverController.js

```diff
@@ -1,24 +1,25 @@
 const controllerCallbacks = new Map();
 
 const contextObserver = new MutationObserver(mutations => {
-	mutations.forEach(mutation => {
-		controllerCallbacks.forEach(callback => callback(mutation.attributeName));
-	});
+	controllerCallbacks.forEach(callback => callback(
+		mutations.map(mutation => mutation.attributeName)
+	));
 });
 
 export class HtmlAttributeObserverController {
 
-	constructor(host, attribute) {
-		this._host = host;
-		this._attribute = attribute;
+	constructor(host, ...attributes) {
+		if (attributes.length === 0) throw new Error(
+			`Can't construct controller; must supply at least one observed attribute.`
+		);
 
-		this.value = undefined;
+		this._host = host;
+		this._attributes = attributes;
+		this.values = new Map();
 	}
 
 	hostConnected() {
-		this.value = document.documentElement.hasAttribute(this._attribute)
-			? document.documentElement.getAttribute(this._attribute)
-			: undefined;
+		this._setAttributes(this._attributes);
```

Comment:
Should the attributes be specified on the constructor or on the `hostConnected` call?  Could a situation arise where we don't know what attributes to observe until after the component is constructed, or we want to observe a different attribute after being reconnected?  Are these valid scenarios?  🤔

Reply:
Hmm, that's a good point. My intent here was to simulate what an actual reactive controller would do (which, as I understand it, would handle the `hostConnected` and `hostDisconnected` under the hood). But I'm also fine changing this behaviour. Certainly right now the attributes we're watching are immutable per controller, but it doesn't have to be that way.